src/types/events-types.ts

```diff
@@ -42,11 +42,39 @@ export type EventsGetReply = GenericMessageReply & {
   cursor?: PaginationCursor;
 };
 
+
+export type EventsSubscribeMessageOptions = {
+  handler: GenericMessageSubscriptionHandler;
+};
+
+export type EventsSubscribeMessage = {
+  authorization: AuthorizationModel;
+  descriptor: EventsSubscribeDescriptor;
+};
+
+export type EventsHandler = (message: GenericMessage) => void;
+
+export type EventsSubscription = {
```

Comment:
Style nit: I think as a convention, when there is method declaration in the typing, we should use `interface` instead of `type, it indicates that this is more than "just" a data holder.